### PR TITLE
Add xPL protocol management

### PR DESCRIPTION
**Pull request by FragJage**
xPL is view as input and output sensors in calaos.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -16,7 +16,7 @@ AC_SYS_LARGEFILE
 LT_PREREQ([2.2])
 LT_INIT([win32-dll])
 
-AM_GNU_GETTEXT_VERSION([0.19])
+AM_GNU_GETTEXT_VERSION([0.18])
```
Comment (raoulh) [inline configure.ac]: Why this change?

Comment (FragJage) [reply configure.ac]: Oups, it's to build on my old ubuntu. Don't include in master branch.

File: src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp [hunk 2]
```diff
@@ -0,0 +1,200 @@
+/*** LICENCE ***************************************************************************************/
```
Comment (naguirre) [inline src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp]: xPLLib seems to be an external Library, it would make sense to put it in lib directory; We did it already for libquickmail or uvw.

Comment (raoulh) [reply src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp]: Actually the wago code does the same, the modbus lib is in IO/Wago

File: src/bin/calaos_server/IO/xPL/xPLController.cpp [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    m_UdpRecvHandle = uvw::Loop::getDefault()->resource<uvw::UDPHandle>();
+
+    m_UdpRecvHandle->on<uvw::UDPDataEvent>([this](const uvw::UDPDataEvent &ev, auto &)
+    {
+        this->udpListenData(ev.data.get(), ev.length, ev.sender.ip, ev.sender.port);
+    });
+
+    m_UdpRecvHandle->on<uvw::ErrorEvent>([this](const uvw::ErrorEvent &ev, uvw::UDPHandle &h)
+    {
+        cErrorDom("network") << "xPL UDP server error : " << ev.what();
```
[the remark this answers is not in the document]
Comment (raoulh) [reply src/bin/calaos_server/IO/xPL/xPLController.cpp]: Or better xpl :)

File: src/bin/calaos_server/IO/xPL/xPLController.cpp [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    if(r!=0) return 3865;
+    return portTCP;  
+}
+
+string xPLController::localAddress()
+{
+    std::vector<uvw::InterfaceAddress> interfaceAdrs;
+    std::vector<uvw::InterfaceAddress>::iterator it;
+  
+    interfaceAdrs = uvw::Utilities::interfaceAddresses();
```
[the remark this answers is not in the document]
Comment (FragJage) [reply src/bin/calaos_server/IO/xPL/xPLController.cpp]: I found nothing in src/lib/Utils.cpp, found getAllInterfaces() in src/lib/tcpspcket.cpp. Is that ?

File: src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp (hunk 2 above)
Comment (FragJage) [reply src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp]: What do you prefers ? If the lib is use only by the IO, it's better to put it in a sub folder of the IO ?

Comment (raoulh) [reply src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp]: I would say keep it like it is, it's ok for me. So the xpl code is at one and only place

**Comment by FragJage** [comment]
All remarks / review processed except functions in calaos::utils don't found or i don't understand naguirre review.

File: src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp (hunk 2 above)
Comment (naguirre) [reply src/bin/calaos_server/IO/xPL/xPLLib/Address.cpp]: yep ok for me too

**Comment by raoulh** [comment]
I think @naguirre was referring to using this code https://github.com/calaos/calaos_base/blob/master/src/lib/tcpsocket.h#L97
instead of `uvw::Utilities::interfaceAddresses()`

I don't know whether the uvw is better or not. Maybe we could improve ours? Actually our is only working on linux. I bet the libuv one is crossplatform...

File: src/bin/calaos_server/IO/xPL/xPLController.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (naguirre) [reply src/bin/calaos_server/IO/xPL/xPLController.cpp]: nevermind, uvw functions seems to be simpler to use and multiplateform. Function in tcpsockets is obsolete now.
And after reading code in tcpsocket.cpp, i think that we could improve this code by using uvw vw::Utilities::interfaceAddresses() :-)
